raiden-ts/src/transfers/utils.ts

```diff
@@ -67,26 +67,30 @@ export function makeMessageId(): UInt<8> {
  */
 export function raidenSentTransfer(sent: SentTransfer): RaidenSentTransfer {
   const [status, changedAt]: [RaidenSentTransferStatus, number] = sent.lockExpiredProcessed
-      ? [RaidenSentTransferStatus.expired, sent.lockExpiredProcessed[0]]
-      : sent.unlockProcessed
-      ? [RaidenSentTransferStatus.unlocked, sent.unlockProcessed[0]]
-      : sent.lockExpired
-      ? [RaidenSentTransferStatus.expiring, sent.lockExpired[0]]
-      : sent.unlock
-      ? [RaidenSentTransferStatus.unlocking, sent.unlock[0]]
-      : sent.secretReveal
-      ? [RaidenSentTransferStatus.revealed, sent.secretReveal[0]]
-      : sent.channelClosed // channelClosed before revealing
-      ? [RaidenSentTransferStatus.closed, sent.channelClosed[0]]
-      : sent.refund
-      ? [RaidenSentTransferStatus.refunded, sent.refund[0]]
-      : sent.transferProcessed
-      ? [RaidenSentTransferStatus.received, sent.transferProcessed[0]]
-      : [RaidenSentTransferStatus.pending, sent.transfer[0]],
-    success: boolean | undefined =
+    ? [RaidenSentTransferStatus.expired, sent.lockExpiredProcessed[0]]
```

Comment:
I get the feeling that we have a few ternaries too much here. Isn't any alternative approach that is elegant and easier to read?

Reply:
Yes, I agree, but I couldn't think on a better alternative yet, because it's not just conditions, but actually a cascading priority order for them. Maybe a separate function can help, but I'm open to suggestions 😄

Reply:
Done =)